File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        res = []
        for num in nums:
            ans = -1
            for j in range(1, num):
                if (j | (j+1)) == num:
                    ans = j
                    break
            res.append(ans)
        return res

File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        for(int &x: nums){
            int res = -1;
            int num = 1;
            while((x & num) != 0){
                res = x - num;
                num <<= 1;
            }
            x = res;
        }
        return nums;
    }
};